### DISCUSSION TITLE
Implement Payment Recovery on custom payment screen in POS

### DISCUSSION THREAD

**Discussion by YoshaPor** [microsoft/Dynamics365Commerce.Solutions#27] [Q&A] [answered]
Hi Team,

We have a customization to make payments with 3rd party financing options (card not present scenario), we have used custom flow to capture the payment on POS and are implementing IOperationRequestHandlerAsync. The system initiates the payment from a custom payment screen.
We need to ensure that through this design, the system checks for recovered payments from credit card if any and prompts the user with the recovered payment.

As per our analysis, CheckForRecoveredPaymentTransactionClientRequest is executed from POS to check for any recovered payments on device and shows the below dialogue to the user if any payments are found:
 

<img width="481" alt="image" src="https://user-images.githubusercontent.com/122339378/211469176-110cc654-0ad6-4687-8a62-94b3dbb308c2.png">

We do not see any extension point for the above request and are unable to invoke it from custom flow.
Any guidance on how payment recovery can be achieved for above will be helpful.

**Comment by madyke** [comment, marked as answer]
Hi @YoshaPor - I confirmed that there are no extension points for that request, and that it is not exposed through Commerce SDK as a consumable request.

From your description, it sounds like what you want is to be able to make your own call to that request and get the response in your custom flow, is that correct? If so, you will need to file an extensibility request to make this request consumable for extensions:
[Extensibility requests - Finance & Operations | Dynamics 365 | Microsoft Learn](https://learn.microsoft.com/en-us/dynamics365/fin-ops-core/dev-itpro/extensibility/extensibility-requests)

**Reply by YoshaPor** [reply]
Hi @madyke - Thanks for looking into it. We have already raised an extensibility request. 
Yes, we are expecting to fetch the Recovered Payments through the custom flow from POS by calling this request.
Since we are implementing IOperationRequestHandlerAsync in CRT, I wanted to check if there is any other way through which we can check for the Recovered Payments from our custom flow?

**Reply by madyke** [reply]
I don't think that the `IOperationRequestHandlerAsync` in CRT will be able to help here. Server-side customizations like that won't affect your ability to make calls using that request on the client-side (POS).

So unfortunately, it won't be possible to create the `CheckForRecoveredPaymentTransactionClientRequest` until we have added it as an extension request that you can consume through the SDK.
